File: engine/app/schemas/mt5.py
from __future__ import annotations

from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel, Field


class Mt5ConnectionState(StrEnum):
    CONNECTED = "connected"
    DISCONNECTED = "disconnected"
    UNAVAILABLE = "unavailable"


class Mt5Diagnostic(BaseModel):
    code: str
    message: str
    remediation: str


class AccountSummary(BaseModel):
    login: int
    server: str
    company: str | None = None
    currency: str
    balance: float
    equity: float


class TerminalSummary(BaseModel):
    path: str | None = None
    version: str | None = None
    connected: bool


class Mt5StatusResponse(BaseModel):
    state: Mt5ConnectionState
    terminal: TerminalSummary | None = None
    account: AccountSummary | None = None
    diagnostic: Mt5Diagnostic | None = None


class SymbolResponse(BaseModel):
    name: str
    description: str
    path: str
    digits: int
    point: float
    visible: bool


class SymbolListResponse(BaseModel):
    items: list[SymbolResponse]
    total: int
    fetched_at: datetime

File: frontend/src/types/mt5.ts
export type Mt5ConnectionState = "connected" | "disconnected" | "unavailable";

export interface Mt5Diagnostic {
  code: string;
  message: string;
  remediation: string;
}

export interface TerminalSummary {
  path: string | null;
  version: string | null;
  connected: boolean;
}

export interface AccountSummary {
  login: number;
  server: string;
  company: string | null;
  currency: string;
  balance: number;
  equity: number;
}

export interface Mt5StatusResponse {
  state: Mt5ConnectionState;
  terminal: TerminalSummary | null;
  account: AccountSummary | null;
  diagnostic: Mt5Diagnostic | null;
}

export interface SymbolResponse {
  name: string;
  description: string;
  path: string;
  digits: number;
  point: number;
  visible: boolean;
}

export interface SymbolListResponse {
  items: SymbolResponse[];
  total: number;
  fetched_at: string;
}
